### PR TITLE
AddEditTask: adds rotation tests and fixes persistance bug

### PR DESCRIPTION
**Pull request by JoseAlcerreca**
Fixing  #217. 

setRetainInstance was not needed, turns out. Not super happy with the `shouldLoadDataFromRepo`. Suggestions?

### CODE REVIEW

**Comment by passsy** [comment]
What if loading the task (could theoretically happen from the database or the network) doesn't finish before the Activity receives the first configuration change? The current implementation wouldn't populate the task again because the fragment is not null although the task isn't loaded. 

The `shouldLoadDataFromRepo` should be set when the presenter has finished loading the task. It should also be saved in `onSaveInstanceState()`.

**Comment by savepopulation** [comment]
I think you should focus on repository part to handle orientation change. When a request already made to get tasks you should not send same request just observe the result.

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java [hunk 1]
```diff
@@ -50,6 +50,9 @@ protected void onCreate(Bundle savedInstanceState) {
         AddEditTaskFragment addEditTaskFragment =
                 (AddEditTaskFragment) getSupportFragmentManager().findFragmentById(R.id.contentFrame);
 
+        // Prevent the presenter from loading data from the repository if this is a config change.
+        boolean shouldLoadDataFromRepo = addEditTaskFragment == null;
```
Comment (slinzner) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java]: This is very cryptic, i'd rather prefer something like this:
boolean doNotLoadDataOnConfigChange = false;
if {null == aETF} { doNotLoadDataOnConfigChange = true}

This saves you the comment and the reader immediately knows what's going on

File: todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksScreenTest.java [hunk 2]
```diff
@@ -33,33 +54,11 @@
...
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
-
 import static com.example.android.architecture.blueprints.todoapp.TestUtils.getCurrentActivity;
 import static com.google.common.base.Preconditions.checkArgument;
-
 import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.core.IsNot.not;
 
-import android.support.test.InstrumentationRegistry;
```
Comment (slinzner) [inline todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksScreenTest.java]: Was this intentionally or did you just use the wrong formatter?

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.java [hunk 3]
```diff
@@ -86,9 +86,8 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         View root = inflater.inflate(R.layout.addtask_frag, container, false);
         mTitle = (TextView) root.findViewById(R.id.add_task_title);
         mDescription = (TextView) root.findViewById(R.id.add_task_description);
-
+        setRetainInstance(false);
```
Comment (slinzner) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.java]: false is the default, no need to explicitly set it here

**Comment by JoseAlcerreca** [comment]
@passsy had to read it 5 times to understand what you meant and you're right! We should cover that case in tests. I'll send another commit soon.

**Comment by passsy** [comment]
I'll use simpler sentences in the future 😊

**Comment by JoseAlcerreca** [comment]
My fault, I thought you were talking about something else. PTAL!

**Comment by passsy** [comment]
LGTM 🚢

**Comment by malmstein** [comment]
LGTM!

**Pull request merged**